src/rcc/mod.rs

```diff
@@ -1,22 +1,46 @@
 //! Reset and Clock Control
 
-use stm32l4x6::{rcc, RCC};
+// TODO right now the various configure functions reach into rcc directly. This is bad. Add an
+// opaque CR member to RCC, and add methods to CR and BDCR. They should probably take clock source
+// variant arguments.
 
-use ::cmp;
+use stm32l4x6::{rcc, PWR, RCC};
 
 use common::Constrain;
-use time::Hertz;
 use flash::ACR;
+use time::Hertz;
+
+pub mod clocking;
 
 impl Constrain<Rcc> for RCC {
+    /// Create an RCC peripheral handle.
+    ///
+    /// Per Reference Manual Ch. 6.2 the default System Clock source is MSI clock with frequency 4 MHz
+    ///
+    /// The `constrain` method enables write access to the BDCR, and the `freeze` method disables
+    /// it again. This is to enable changing LSE- and RTC-related settings.
     fn constrain(self) -> Rcc {
+        // Enable write access to the BDCR; this is necessary to enable the LSE and change RTC
+        // settings.
+        unsafe {
```

Comment:
There is https://github.com/DoumanAsh/stm32l4x6_hal/blob/master/src/power.rs#L45 for this purpose

Also note afaik you  need to wait for BDCR lock to be lifted(it should be few ticks, but it is still better to wait)